src/main/java/teammates/ui/webapi/WebApiServlet.java

```diff
@@ -149,4 +152,28 @@ private void throwError(HttpServletResponse resp, int statusCode, String message
         result.send(resp);
     }
 
+    private void putUserInfo(Map<String, Object> responseDetails, Action action) {
+        if (action == null) {
+            return;
+        }
+
+        String googleId = action.userInfo == null ? null : action.userInfo.getId();
+        String regkey = action.getRequestParamValue(Const.ParamsNames.REGKEY);
+        String studentEmail = action.getRequestParamValue(Const.ParamsNames.STUDENT_EMAIL);
+
+        if (regkey != null && studentEmail == null) {
+            StudentAttributes student = action.logic.getStudentForRegistrationKey(regkey);
```

Comment:
By right logic should not be visible/callable here. These user-related information needs to be persisted within the action class itself and obtained from there.

Reply:
This is in the `finally` block so the action should finish executing and result is already sent so it is true that the user-related information is already persisted at that point of time?

Reply:
Then why not just get the information from `action.getSomething()` instead of going through the `logic` field?